queries/fennel/highlights.scm

```diff
@@ -46,80 +36,217 @@
 (list
   .
   (multi_symbol
-    (symbol) @function.call .))
+    member: (symbol_fragment) @function.call .))
 
-((symbol) @variable.builtin
-  (#lua-match? @variable.builtin "^[$]"))
-
-(binding) @string.special.symbol
-
-[
-  "fn"
-  "lambda"
-  "hashfn"
-  "#"
-] @keyword.function
-
-(fn
-  name:
-    [
-      (symbol) @function
-      (multi_symbol
-        (symbol) @function .)
-    ])
-
-(lambda
-  name:
-    [
-      (symbol) @function
-      (multi_symbol
-        (symbol) @function .)
-    ])
-
-[
-  "for"
-  "each"
-] @keyword.repeat
+(multi_symbol_method
+  ":" @punctuation.delimiter
+  method: (symbol_fragment) @function.method.call .)
+
+; Just `&` is only used in destructuring
+((symbol) @punctuation.special
+  (#eq? @punctuation.special "&"))
+
+; BUG: It should only be valid if inside hashfn of any depth, but
+; it's impossible to express such query at the moment of writing.
+; See tree-sitter/tree-sitter#880
+((symbol) @variable.parameter
+  (#vim-match? @variable.parameter "^\\$(...|[1-9])$"))
```

Comment:
Is a full-on regex necessary here, or can you do this with (possibly multiple) Lua patterns?

Reply:
I wanted to do a Lua regex, but it didn't have the `|` operator. The regex should only match either for `$...` or `$[1-9]`, but having multiple Lua regex means it has to be **both** `$...` and `$[1-9]`

Reply:
Done!

Reply:
Nice!